src/constructs/named-backreference.ts

```diff
@@ -0,0 +1,24 @@
+//import { encodeSequence } from '../encoder/encoder';
+import type { EncodeResult } from '../encoder/types';
+//import { ensureArray } from '../utils/elements';
+import type { RegexConstruct } from '../types';
+
+export interface NamedBackreference extends RegexConstruct {
+  type: 'named-backreference';
+  name: string;
+}
+
+export function namedBackreference(groupName: string): NamedBackreference {
```

Comment:
In #66 I proposed an option to skip the `name`, in which case Regex Builder would autogenerate a brief name for the user (`ref-1`, etc). That matches Swift Regex Builder which defines `Reference` without name parameter at all. Not sure if this is worth it. wdyt?